### ISSUE TITLE
Does `nupac upgrade` properly handle nupac's upgrade after breaking changes?

### ISSUE THREAD

**Issue by skelly37** [skelly37/nupac#98]
### Describe the issue

- [ ] Check if the current `nupac upgrade` procedure handles it properly: basically we should depend only on the `version` field in this case.
- [ ] Assure that nupac package is always updated first
- [ ] Find a way to block the further installations/upgrades after finding out that major has been incremented (maybe verify current nupac's version with the one in repo?) **`nupac version` needs refactoring**. As long as there is diff between nupac and the json, block actions.

If not, the procedure should look that way:
1) update the repo cache
2) fetch metadata json
3) verify json's checksum with the one in the repo cache
4) fetch nupac.nu
5) verify nupac.nu's checksum with the one in the json
6) if checksums are correct, replace metadata and .nu in nupac/packages/nupac
7) prompt the user to restart the shell or to `use` nupac again

Basically nupac should be immune to breaking changes in the jsons, so it gets upgraded and lets users use it once again.

### Argumentation

We do not want users to re-run installer.nu or (even worse) download source after incrementing major
